### DISCUSSION TITLE
How do you index strings longer than 32 characters?

### DISCUSSION THREAD

**Discussion by PaulZhemanov** [FuelLabs/fuel-indexer#1233] [Q&A] [answered]
Hi. How do you index strings longer than 32 characters?

**Comment by ra0x3** [comment, marked as answer]
@PaulZhemanov We use the `Charfield` scalar for indexing strings.

```graphql 
type Person {
  id: ID!
  name: Charfield! # Can be up to 255 characters long
}
```

In your indexer handler you can convert `SizedAsciiString` (the SDK's string type) to indexer string types by just doing

```rust 
mod my_module {
  fn my_handler(event: MyEvent) {
    // Assuming event.name is a str[32] defined in your contract 
    let p = Person::new(event.name.to_string());
  }
}
```

Let me know if this helps. Also gonna move this to a discussion :)

**Reply by ra0x3** [reply]
@PaulZhemanov Checking in to see if that worked for you?

**Reply by PaulZhemanov** [reply]
Yes, it works . Thanks🤗
